### PR TITLE
fix(es/compat): Preserve more class names

### PR DESCRIPTION
**Pull request by magic-akari**
**Description:**
Note: We do not handle some class name which is not valid ident as following .
```JavaScript
const foo = {
     1: class {},
     "-": class {},
}
```

**BREAKING CHANGE:**

**Related issue (if exists):**
- #5102 
- #5104

### CODE REVIEW

File: crates/swc_ecma_transforms_compat/src/es2015/classes/mod.rs [hunk 1]
```diff
@@ -260,17 +261,81 @@ where
...
+        }
+
+        n.visit_mut_children_with(self)
+    }
+
     fn visit_mut_assign_expr(&mut self, a: &mut AssignExpr) {
         if let AssignExpr {
-            op: op!("="),
-            left: PatOrExpr::Pat(pat),
+            op: op!("=") | op!("||=") | op!("??="),
```
[the remark this answers is not in the document]
Comment (magic-akari) [reply crates/swc_ecma_transforms_compat/src/es2015/classes/mod.rs:321]: check this.
https://github.com/swc-project/swc/blob/13ea5a71fcc65d26d07f91b808fcf398101e597a/crates/swc/tests/fixture/issues-5xxx/5102/input/index.js#L12-L13
https://github.com/swc-project/swc/blob/13ea5a71fcc65d26d07f91b808fcf398101e597a/crates/swc/tests/fixture/issues-5xxx/5102/output/index.js#L53-L60

[the remark this answers is not in the document]
Comment (Austaras) [reply crates/swc_ecma_transforms_compat/src/es2015/classes/mod.rs:321]: It gets transformed by es2021 into first
```js
b || (b = class {
});
c ?? (c = class {
});
```

[the remark this answers is not in the document]
Comment (magic-akari) [reply crates/swc_ecma_transforms_compat/src/es2015/classes/mod.rs:321]: You are right.
I don't know if anyone just use `es2015::classes` without es2021 pass.
And I update the correct test case now.

[the remark this answers is not in the document]
Comment (magic-akari) [reply crates/swc_ecma_transforms_compat/src/es2015/classes/mod.rs:321]: @kdy1 
Should we revert to this change and ignore `||=` and `??=`?  Or should we handle it?

[the remark this answers is not in the document]
Comment (kdy1) [reply crates/swc_ecma_transforms_compat/src/es2015/classes/mod.rs:321]: Hmm, I think it's better to handle it, as we don't know all use cases.

**Review by kdy1 — APPROVED** [review]
Thank you!

---

swc-bump:
 - swc_ecma_transforms_compat

**Pull request merged**